internal/services/dedicatedcloud/dedicatedcloud.go

```diff
@@ -6,23 +6,649 @@ package dedicatedcloud
+		localFilers, err := httpLib.FetchExpandedArray(localFilersEndpoint, "")
+		if err != nil {
+			localFilers = []map[string]any{}
+		}
+
+		globalFilersEndpoint := fmt.Sprintf("/dedicatedCloud/%s/filer", url.PathEscape(args[0]))
+		globalFilers, err := httpLib.FetchExpandedArray(globalFilersEndpoint, "")
+		if err != nil {
+			globalFilers = []map[string]any{}
+		}
+		allFilers := append(localFilers, globalFilers...)
+
+		// Calculate totals
+		totalCores := 0
+		totalRAM := 0.0
+		totalVMs := 0
+		totalDiskSpace := 0.0
+
+		// Sum from hosts
+		for _, host := range hosts {
+			// Sum cores
+			if cpuNumRaw, ok := host["cpuNum"]; ok && cpuNumRaw != nil {
+				totalCores += toInt(cpuNumRaw)
+			}
+
+			// Sum RAM
+			if ram, ok := host["ram"].(map[string]any); ok {
+				totalRAM += toFloat64(ram["value"])
+			}
+
+			// Sum VMs
+			if vmTotal, ok := host["vmTotal"]; ok && vmTotal != nil {
+				switch v := vmTotal.(type) {
+				case float64:
+					totalVMs += int(v)
+				case json.Number:
+					if i, err := v.Int64(); err == nil {
+						totalVMs += int(i)
+					}
+				}
+			}
+		}
+
+		// Sum disk space from filers
+		for _, filer := range allFilers {
+			if size, ok := filer["size"].(map[string]any); ok {
+				if sizeValueRaw, ok := size["value"]; ok && sizeValueRaw != nil {
+					sizeValue := toFloat64(sizeValueRaw)
+					// Convert to GB if needed
+					if sizeUnit, ok := size["unit"].(string); ok {
+						if strings.ToUpper(sizeUnit) == "TB" {
+							sizeValue *= 1000 // Convert TB to GB
+						} else if strings.ToUpper(sizeUnit) == "MB" {
+							sizeValue /= 1000 // Convert MB to GB
+						}
+					}
+					totalDiskSpace += sizeValue
+				}
+			}
+		}
+
+		// Add totals to datacenter
+		datacenters[i]["totalCores"] = totalCores
+		datacenters[i]["totalRAM"] = fmt.Sprintf("%.0f GB", totalRAM)
+		datacenters[i]["totalVMs"] = totalVMs
+		datacenters[i]["totalDiskSpace"] = fmt.Sprintf("%.0f GB", totalDiskSpace)
+	}
+
+	// Filter results
+	datacenters, err = filters.FilterLines(datacenters, flags.GenericFilters)
+	if err != nil {
+		display.OutputError(&flags.OutputFormatConfig, "failed to filter results: %s", err)
+		return
+	}
+
+	// Display table
+	display.RenderTable(datacenters, []string{"datacenterId", "name", "version", "commercialName", "totalCores", "totalRAM", "totalVMs", "totalDiskSpace"}, &flags.OutputFormatConfig)
+}
+
+func GetDatacenter(_ *cobra.Command, args []string) {
+	getDatacenterWithOptions(args, false, false, false, true)
+}
+
+func GetDatacenterHosts(_ *cobra.Command, args []string) {
+	getDatacenterWithOptions(args, true, false, false, false)
+}
+
+func GetDatacenterFilers(_ *cobra.Command, args []string) {
+	getDatacenterWithOptions(args, false, true, false, false)
+}
+
+func GetDatacenterClusters(_ *cobra.Command, args []string) {
+	getDatacenterWithOptions(args, false, false, true, false)
+}
+
+func getDatacenterWithOptions(args []string, includeHosts, includeFilers, includeClusters, includeTotals bool) {
```

Comment:
I think this func and the related CLI commands should be re-designed a bit:
- the other commands of the CLI are designed to fetch one kind of object per command. Here you have 4 commands that fetch the same base and then other sub-objects. It increases the number of HTTP calls necessary to run any command, so it will be longer to execute and more error-prone in case of API errors. I would prefer that each command only fetches one kind of information (only the hosts, only the filers, ...), OR having only the command to fetch a datacenter with flags like `--with-hosts`, `--with-filers`, …
- this single func to fetch everything is hard to read so it will be hard to maintain, it should at least be splitted in separate funcs